server/pxf-hdfs/src/main/java/org/greenplum/pxf/plugins/hdfs/LineBreakAccessor.java

```diff
@@ -129,7 +128,25 @@ public boolean writeNextObject(OneRow onerow) throws IOException {
             // match same buffer size to read data from the input stream. The
             // buffer size can be configured externally.
             int bufferSize = configuration.getInt("io.file.buffer.size", DEFAULT_BUFFER_SIZE);
-            long count = IOUtils.copyLarge((InputStream) onerow.getData(), dos, new byte[bufferSize]);
+
+            long count = 0;
+
+            // Copying the logic from IOUtils.copyLarge to here to add logging
+            // In case things go wrong, this logging will help with further investigation.
+            InputStream inputStream = (InputStream) onerow.getData();
+            final byte[] buffer = new byte[bufferSize];
+            final int EOF = -1;
+            int n, byteCount;
+            while (EOF != (n = inputStream.read(buffer))) {
+                dos.write(buffer, 0, n);
+                byteCount = n;
+                if(byteCount % 2*DEFAULT_BUFFER_SIZE == 0){
+                    // Log this message after every 2*8K Bytes
+                    LOG.trace("Successfully Wrote {} bytes to outputStream using a buffer of size {}", byteCount, bufferSize);
+                }
+                count += n;
+            }
```

Comment:
great, but let's re-arrange a bit, get rid of new variable `byteCount`, adjust logged message, do calculations only if `LOG.trace` is enabled and choose a window that will log about every 5 secs. We have seen the upload speed of 8MB/sec, so like 1000 buffers per sec or 5000 buffers per 5 secs. Otherwise the log will be littered with entries. Or we can introduce a property that can manage the frequency.
```suggestion
            InputStream inputStream = (InputStream) onerow.getData();
            final byte[] buffer = new byte[bufferSize];
            final int frequency = 5000 * bufferSize;

            // the logic below is copied from IOUtils.copyLarge to add logging
            long count = 0;            
            int n;
            while (-1 != (n = inputStream.read(buffer))) {
                dos.write(buffer, 0, n);
                count += n;
                if (LOG.isTraceEnabled() && n % frequency == 0) {
                    // log this message after every frequency*8K (default) bytes
                    LOG.trace("wrote {} additional bytes, total so far {} (buffer size {})", n, count, bufferSize);
                }
            }
```

Reply:
Also, since we will be printing not for every row, maybe using DEBUG level here will be more appropriate